File: sdk/src/main/java/com/akkaserverless/javasdk/EntityContext.java
/*
 * Copyright 2019 Lightbend Inc.
 */

package com.akkaserverless.javasdk;

/**
 * Root context for all contexts that pertain to entities, that is, things that are addressable via
 * an entity id.
 */
public interface EntityContext extends Context {

  /**
   * The id of the entity that this context is for.
   *
   * @return The entity id.
   */
  String entityId();
}

File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/EntityContext.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk

/**
 * Root context for all contexts that pertain to entities, that is, things that are addressable via an entity id.
 */
trait EntityContext extends Context {

  /**
   * The id of the entity that this context is for.
   *
   * @return
   *   The entity id.
   */
  def entityId: String
}
